code/__HELPERS/lists.dm

```diff
@@ -405,6 +405,18 @@
 		if(islist(.[i]))
 			.[i] = .(.[i])
 
+//takes an input_key, as text, and the list of keys already used, outputting a replacement key in the format of "[input_key] ([number_of_duplicates])" if it finds a duplicate
+//use this for lists of things that might have the same name, like mobs or objects, that you plan on giving to a player as input
+/proc/avoid_assoc_duplicate_keys(input_key, list/used_key_list)
+	if(!input_key || !istype(used_key_list))
+		return
+	if(input_key in used_key_list)

```

Comment:
Don't use `in` with associated lists, it has to transverse the entire list to check. (O(n))

the faster less wasteful way is to just try to access the key, since it's O(log n) and returns null if the key doesn't exist